### ISSUE TITLE
Test that arguments given to spawnProcess arrive intact

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4079]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)| @jonathanj reported|
|-|-|
|Trac ID|trac#4079|
|Type|enhancement|
|Created|2009-11-10 21:14:01Z|
|Branch|https://github.com/twisted/twisted/tree/test-spawnprocess-arguments-4079|

In the "win32-pipe-quoting-3933-2" branch, a useful test was introduced, `test_processCommandLineArguments`, which verified that arguments passed to `spawnProcess` arrived intact. [#3933](https://github.com/twisted/twisted/issues/3933) has since been marked invalid, it would still be a good idea to merge this test.

<details><summary>Searchable metadata</summary>

```
trac-id__4079 4079
type__enhancement enhancement
reporter__jonathanj jonathanj
priority__normal normal
milestone__ 
branch__branches_test_spawnprocess_arguments_4079 branches/test-spawnprocess-arguments-4079
branch_author__jonathanj jonathanj
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1257887641000000 1257887641000000
changetime__1258288209000000 1258288209000000
version__None None
owner__jonathanj jonathanj
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_2"></a>|@jonathanj commented|
|-|-|

(In [27447]) Branching to 'test-spawnprocess-arguments-4079'

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)|@jonathanj removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @jonathanj|
|-|-|

Just one comment, although important:
```
+            d.addBoth(lambda dummy: reactor.stop())
```

It doesn't work as expected, because it swallows any error the test may raise. Something like that should be done:

```
        def shutdown(passthrough):
            reactor.stop()
            return passthrough

        def spawnChild():
            d = succeed(None)
            d.addCallback(lambda dummy: utils.getProcessOutput(
                sys.executable, ['-c', source] + args, reactor=reactor))
            d.addCallback(processFinished)
            d.addBoth(shutdown)
```

Once done, please merge if windows buildbots are happy. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_6"></a>|@jonathanj commented|
|-|-|

(In [27494]) Don't swallow test errors. refs [#4079](https://github.com/twisted/twisted/issues/4079)

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_7"></a>|@jonathanj set status to `closed`|
|-|-|

(In [27495]) Merge test-spawnprocess-arguments-4079.

Author: jonathanj
Reviewer: therve
Fixes: [#4079](https://github.com/twisted/twisted/issues/4079)

Add a test to verify that arguments passed to spawnProcess arrive as originally
intended at the child process.
